src/openvic-simulation/research/Invention.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "openvic-simulation/economy/BuildingType.hpp"
+#include "openvic-simulation/military/Unit.hpp"
+#include "openvic-simulation/misc/Modifier.hpp"
+#include "openvic-simulation/types/IdentifierRegistry.hpp"
+#include <string_view>
+#include <unordered_set>
+
+namespace OpenVic {
+	struct Invention : Modifier {
+		friend struct InventionManager;
+		//TODO implement limit and chance
+		using unit_set_t = std::unordered_set<Unit const*>;
+		using building_set_t = std::unordered_set<BuildingType const*>;
+		using crime_set_t = std::unordered_set<Crime const*>;
+
+	private:
+		const bool PROPERTY_CUSTOM_PREFIX(news, is);
+		unit_set_t PROPERTY(activated_units);
+		building_set_t PROPERTY(activated_buildings);
+		crime_set_t PROPERTY(enabled_crimes);
```

Comment:
This needs to be added to the `Invention` constructor and `InventionManager::add_invention`, then passed to it in `InventionManager::load_inventions_file`.